User guide documentation

MayaGans opened the pull request:
- adding colors _directly_ to assets.css because the color did not change in custom
- adding categories for each block (we might want to change file naming convention)

- need to add documentation for each block: 
  - block's purpose
  - arguments it takes on
  - stats example

MayaGans committed:
Merge pull request #98 from gvwilson/saving-state

Saving state

MayaGans committed:
Merge pull request #99 from gvwilson/ungroup-block

Implementing an 'ungroup' block.

MayaGans committed:
Merge pull request #100 from gvwilson/95-reverse-block

Reverse the order of rows

MayaGans committed:
Merge https://github.com/tidyblocks/tidyblocks

MayaGans committed:
Merge branch 'master' of https://github.com/tidyblocks/tidyblocks

MayaGans committed:
fixing typo

MayaGans committed:
Merge https://github.com/tidyblocks/tidyblocks

MayaGans committed:
custom css_

MayaGans committed:
add block categories

MayaGans committed:
add block color theme in assets because custom will not override

MayaGans committed:
adding data block documentation

MayaGans committed:
finished data block guide

MayaGans committed:
reformatting data blocks

gvwilson commented on _posts/2019-08-25-License.md:
```diff
@@ -0,0 +1,31 @@
+---
```
Can this please stay in the root directory as LICENSE.md ? That's the Git/GitHub conventional location for a license file, and other pages in the `guide` directory can uplink to it.

gvwilson commented on _posts/2019-08-28-ggplot.md:
```diff
@@ -0,0 +1,11 @@
+---
+title: 'GGPLOT Blocks'
```
Do you want to keep the names `ggplot` and `dplyr` (which won't make any sense at all to our intended audience, who haven't seen R)? Alternatives would be `plotting` and... um... OK, I don't know what a good name for the data manipulation blocks would be.

gvwilson commented on _posts/2019-08-29-dplyr.md:
```diff
@@ -0,0 +1,43 @@
+---
+title: 'DPLYR Blocks'
```
As above regarding the name `dplyr`.

gvwilson commented on _posts/2019-08-29-dplyr.md:
```diff
@@ -0,0 +1,43 @@
+---
+
+
+<img src="/guide/images/dplyr_filter.png" width="30%">
```
This is a case where I think the URL has to be run through `relative_url` *or* the link has to be relative e.g. `src="../images/dplyr_filter.png"` in order to work everywhere.

gvwilson commented on _posts/2019-08-29-dplyr.md:
```diff
@@ -0,0 +1,43 @@
+Logical predicates defined in terms of the variables within the data block. Multiple conditions are combined with AND blocks. Only rows where the condition evaluates to TRUE are kept.
+
+### Examples:
+<br>
```
Why the `<br>` ?

gvwilson commented on _posts/2019-08-29-dplyr.md:
```diff
@@ -0,0 +1,43 @@
+
+### Description:
+
+Most data operations are done on groups defined by variables. The group by block takes an existing table and converts it into a grouped table where operations are performed "by group". The ungroup() block can be used to remove grouping.
```
This is a really good description - thanks!

gvwilson commented on _posts/2019-08-30-data.md:
```diff
@@ -0,0 +1,79 @@
+layout: nil
+---
+
+### All data blocks are hat blocks; you cannot place a block on top of a data block but you can perform data transformations and visualizations on data blocks by connecting blocks under them. 
```
Will our intended audience understand "hat" block?

gvwilson commented on static/assets.css:
```diff
@@ -1,3 +1,13 @@
+footer {
```
I'll see if I can get this to load from the `custom.css` file

gvwilson commented on tidyblocks/gui.js:
```diff
@@ -74,7 +74,7 @@ const setUpBlockly = () => {
   )
 
-  TidyBlocksWorkspace.addChangeListener(Blockly.Events.disableOrphans)
+  //TidyBlocksWorkspace.addChangeListener(Blockly.Events.disableOrphans)
```
Whoops: why disable orphans, and why is this in this PR? (It isn't documentation...)

gvwilson commented on tidyblocks/tidyblocks.js:
```diff
@@ -169,7 +169,7 @@ const tbMedian = (values) => {
 
 /**
- * Find median value.
+ * Find minimum value.
```
Whoops - why is this in this PR? (It's a good fix, it just doesn't belong in the same branch as the user guide changes.)

MayaGans committed:
begin dplyr documentation

MayaGans committed:
more dplyr documentation and images

MayaGans committed:
ggplot blocks

MayaGans committed:
plumbing blocks

MayaGans committed:
img path changes

gvwilson committed:
Minor tidying up.

1.  Fixed an image link: `guide/images` should be `/guide/images` with a leading `/`.
2.  Changed layout to `null` (instead of `nil`).

MayaGans committed:
Merge pull request #1 from gvwilson/user-guide-documentation

Minor tidying up.

MayaGans committed:
adding more doc images

MayaGans committed:
rename dplyr

MayaGans committed:
add plotting blocks

MayaGans committed:
plumbing documentation

MayaGans committed:
value documentation

MayaGans committed:
more value documentation
